internal/versions/versions.go

```diff
@@ -240,33 +240,31 @@ func Latest(plugin plugins.Plugin, query string) (version string, err error) {
 	var stdErr strings.Builder
 
 	err = plugin.RunCallback("latest-stable", []string{query}, map[string]string{}, &stdOut, &stdErr)
-	if err != nil {
-		if _, ok := err.(plugins.NoCallbackError); !ok {
-			return version, err
-		}
-
-		allVersions, err := AllVersionsFiltered(plugin, query)
-		if err != nil {
-			return version, err
-		}
-
-		versions := filterOutByRegex(allVersions, numericStartFilterRegex, true)
-		versions = filterOutByRegex(versions, latestFilterRegex, false)
-
+	if err == nil {
```

Comment:
This approach will result in all errors from "latest-stable" falling back to the "all versions" approach. Not just cases where the file/script isn't present. I'd suggest either adding a `HasCallback` method to determine which script/approach to rely on or making this decision on a "file not found" error response from `RunCallback`.

Reply:
I mentioned this in the summary of the PR. This is already the case, so I didn't want to tackle it on this change specifically. I just want to close https://github.com/augustobmoura/asdf-protonge/issues/4. I did reversed the if condition to make it more explicit that it fallbacks to `list-all` ordering, the code is otherwise unchanged aside from the removal of filters when latest-stable is called.

Adding `HasCallback` is also out of scope for this PR

Reply:
ah sorry I should I read your description more thoroughly! Actually I think the behavior that's here now aligns with what was in the bash script version since it only falls through to the "all versions" implementation when a `plugins.NoCallbackError` error is returned which is only the case if "latest-stable" is missing.

Reply:
A test for this case would be helpful to prevent a regression in the future.

Reply:
Ah, I didn't notice the if condition for `NoCallbackError` (well, I did notice it but couldn't understand that it is a type check). So yeah, the code should have the same behavior as the bash version now